lib/devise/jwt/revocation_strategies/whitelist.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'active_support/concern'
+
+module Devise
+  module JWT
+    module RevocationStrategies
+      # This strategy must be included in the user model, and requires that it
+      # has a :has_many :whitelisted_jwts association.
+      # The JwtWhitelist table must include `jti`, `aud` and `user_id` columns
+      #
+      # In order to tell whether a token is revoked, it just tries to find the
+      # `jti` and `aud` values from the token on the `whitelisted_jwts`
+      # table for the respective user.
+      #
+      # If the values don't exist means the token was revoked.
+      # On revocation, it deletes the matching record from the 
+      # `whitelisted_jwts` table.
+      # On sign in, it creates a new record with the `jti` and `aud` values.
+      module Whitelist
+        extend ActiveSupport::Concern
+
+        included do
+          has_many :whitelisted_jwts, dependent: :destroy
+
+          # @see Warden::JWTAuth::Interfaces::RevocationStrategy#jwt_revoked?
+          def self.jwt_revoked?(payload, user)
+            !user.whitelisted_jwts.exists?(payload.slice('jti', 'aud'))
+          end
+
+          # @see Warden::JWTAuth::Interfaces::RevocationStrategy#revoke_jwt
+          def self.revoke_jwt(payload, user)
+            user
+              .whitelisted_jwts
+              .find_or_initialize_by(payload.slice('jti', 'aud'))
```

Comment:
Just a question on this. Why do we need to initialize if the token does not exist? Is it because `destroy` would just return `false` for a not persisted record? If so, I think it would be better to just `find_by` and `destroy!`, forcing that if something unexpected happens along the way an exception is raised (if a token is given to the server in order to be revocated and it doesn't have it in the database, something unexpected happened, so better to have it logged though the exception).

Reply:
Yes that might be better, i'll update it